include/fuzzuf/mutator/mutator.hpp

```diff
@@ -306,13 +306,21 @@ void Mutator<Tag>::Havoc(u32 stacking, const std::vector<AFLDictData> &extras,
   fuzzuf::optimizer::Store::GetInstance().Set(
       fuzzuf::optimizer::keys::AutoExtras, AFLDictRef(a_extras));
 
+  bool useSelectedCaseHistogram =
+      fuzzuf::optimizer::Store::GetInstance().Exists(
+          fuzzuf::optimizer::keys::SelectedCaseHistogram);
+
+  std::array<u32, fuzzuf::mutator::NUM_CASE> selected_case_histogram;
+  selected_case_histogram.fill(0);
+
   for (std::size_t i = 0; i < stacking; i++) {
     u32 r = mutop_optimizer.CalcValue();
 
     switch (r) {
       case FLIP1:
         /* Flip a single bit somewhere. Spooky! */
         FlipBit(UR(len << 3), 1);
+        selected_case_histogram[r] += 1;
```

Comment:
If you do `selected_case_histogram[r] += 1;` in every case, why not move it to the top of this selected area?

Reply:
It should not be incremented under some conditions and cases like `FLIP16`, `CLONE_BYTES` etc.
https://github.com/fuzzuf/fuzzuf/blob/144645f07ba1893335536a406347bfe505b5f62b/include/fuzzuf/mutator/mutator.hpp#L356
https://github.com/fuzzuf/fuzzuf/blob/144645f07ba1893335536a406347bfe505b5f62b/include/fuzzuf/mutator/mutator.hpp#L528-L551

This is from original MOpt-AFL
https://github.com/puppet-meteor/MOpt-AFL/blob/master/MOpt/afl-fuzz.c#L8351-L8396

but not sure if there is such a case actually not incrementing `selected_case_histogram`

Reply:
I got it. Then this seems legit. I'm now feeling switch-case itself is inconvenient.